internal/config/config.go

```diff
@@ -358,12 +361,21 @@ func Default() *Config {
 		// AWS will have aws-sev-snp as attestation variant
 		Attestation: AttestationConfig{
 			AWSNitroTPM:        &AWSNitroTPM{Measurements: measurements.DefaultsFor(cloudprovider.AWS, variant.AWSNitroTPM{})},
-			AzureSEVSNP:        DefaultForAzureSEVSNP(),
+			AzureSEVSNP:        azureSEVSNP,
 			AzureTrustedLaunch: &AzureTrustedLaunch{Measurements: measurements.DefaultsFor(cloudprovider.Azure, variant.AzureTrustedLaunch{})},
 			GCPSEVES:           &GCPSEVES{Measurements: measurements.DefaultsFor(cloudprovider.GCP, variant.GCPSEVES{})},
 			QEMUVTPM:           &QEMUVTPM{Measurements: measurements.DefaultsFor(cloudprovider.QEMU, variant.QEMUVTPM{})},
 		},
+	}, nil
+}
+
+// DefaultWithPanic returns a struct with the default config.
+func DefaultWithPanic() *Config {
```

Comment:
[Don't panic](https://go-proverbs.github.io/)

We try to avoid using panics. There are rare occasions where it is necessary (regexp.MustCompile), but in these cases the code must panic immediately at startup, and thus be global state. I don't see a need to panic when creating a default config.

Reply:
We don't differentiate code quality of production and test code.

Maybe something that fills default values into a struct shouldn't return an error in the first place.

Reply:
If this function is only used in unit tests, it shouldnt make real API calls (as it is done when calling `DefaultForAzureSEVSNP()`)

I think a better solution for this would be rewriting `DefaultForAzureSEVSNP()` to return some placeholder values for the variables that can be dynamically fetched from our API.
These placeholders should marshal to `"latest"` when we write the config to disk.